providers/vsphere/provider/provider.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package provider
+
+import (
+	"errors"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"go.mondoo.com/cnquery/providers-sdk/v1/vault"
+
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/providers-sdk/v1/upstream"
+	"go.mondoo.com/cnquery/providers/vsphere/connection"
+	"go.mondoo.com/cnquery/providers/vsphere/resources"
+)
+
+type Service struct {
+	runtimes         map[uint32]*plugin.Runtime
+	lastConnectionID uint32
+}
+
+func Init() *Service {
+	return &Service{
+		runtimes:         map[uint32]*plugin.Runtime{},
+		lastConnectionID: 0,
+	}
+}
+
+func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error) {
+	flags := req.Flags
+	if flags == nil {
+		flags = map[string]*llx.Primitive{}
+	}
+
+	conn := &inventory.Config{
+		Type: req.Connector,
+	}
+
+	// parse args
+	user := ""
+	port := 0
+	if len(req.Args) != 0 {
```

Comment:
What happens when `len` is zero?

Reply:
For the cli parsing, if there are no args provided, then the connection will not have any information about it. The caller can still work around this fact and insert their own asset information, but for now we don't do that.